lib/Module/Optimize.cpp

```diff
@@ -80,7 +84,11 @@ static cl::alias A1("S", cl::desc("Alias for --strip-debug"),
 
 // A utility function that adds a pass to the pass manager but will also add
 // a verifier pass after if we're supposed to verify.
+#if LLVM_VERSION_CODE >= LLVM_VERSION(3, 7)
+static inline void addPass(legacy::PassManager &PM, Pass *P) {
```

Comment:
I think we can do a bit cleaner. We can introduce our own header (e.g. `include/klee/Internal/Module/LLVMPassManager.h`) that adds a typedef for us and includes the right header file based on LLVM version

```c++
#if LLVM_VERSION_CODE >= LLVM_VERSION(3,7)
#include "llvm/IR/LegacyPassManager.h"
#else
#include "llvm/PassManager.h"
#endif

// TODO: Check that this namespacing is the right way to do things.
namespace klee {
#if LLVM_VERSION_CODE >= LLVM_VERSION(3,7)
typedef llvm::legacyPassManager LegacyLLVMPassManagerTy;
#else
typedef llvm::PassManager LegacyLLVMPassManagerTy;
#endif
}
```